lib/frame_mach.ml

```diff
@@ -1,12 +1,17 @@
 module I386 = struct
   type t =
     | Unknown
-    | I386          [@value 1]
+    | I386
     | I8086
     | I386_intel
-    | X86_64        [@value 64]
+    | X86_64
     | X86_64_intel
-    [@@deriving enum]
+  [@@deriving enumerate]
+
+  include Frame_enum.Make_substitute(struct
```

Comment:
that substitution is not correct, in the `enum` deriver, the value stanza resets an internal counter in a such way, that all consequent constructors are numbered starting with the new value of a counter. In your particular example, the value of `x86_64_intel` is `5` while it should `65`.

Reply:
The same is applicable to all other cases.